File: Data Structures & Algorithms/count-vowel-strings-in-ranges/submission-0.py
class Solution:
    def vowelStrings(self, words, queries):

        vowels = {'a', 'e', 'i', 'o', 'u'}

        prefix = [0]

        for word in words:
            if word[0] in vowels and word[-1] in vowels:
                prefix.append(prefix[-1] + 1)
            else:
                prefix.append(prefix[-1])

        ans = []

        for l, r in queries:
            ans.append(prefix[r + 1] - prefix[l])

        return ans

File: Data Structures & Algorithms/count-prefix-and-suffix-pairs-i/submission-0.cpp
class Solution {
public:
    int countPrefixSuffixPairs(vector<string>& words) {

        int ans = 0;

        for (int i = 0; i < words.size(); i++) {

            for (int j = i + 1; j < words.size(); j++) {

                string a = words[i];
                string b = words[j];

                if (b.substr(0, a.size()) == a &&
                    b.substr(b.size() - a.size()) == a) {

                    ans++;
                }
            }
        }

        return ans;
    }
};